manim/mobject/types/vectorized_mobject.py

```diff
@@ -2051,15 +2058,42 @@ def construct(self):
                         (gr-circle_red).animate.shift(RIGHT)
                     )
         """
+
+        flattened_args = []
+
         for m in vmobjects:
-            if not isinstance(m, (VMobject, OpenGLVMobject)):
+            # Mobject and its subclasses are iterable
+            if not isinstance(m, (Iterable, Generator)):
```

Comment:
Perhaps I may not have been clear enough in my last review, but you could reduce the amount of explanation needed and the clarity of the code by doing the following:
```py
if isinstance(mob, VMobject): # first
    ...
elif isinstance(mob, Iterable): # generator not needed, it's a subprotocol of Iterable
    for submob in mob:
        if not isinstance(submob, VMobject):
            raise
    flattened.extend(mob)
else:
    raise
```
The benefit of this is that it's easier to read and it avoids repeating the same text for the `ValueError` in 3 places (it repeats it twice, in the ideal case it should never repeat it)

Reply:
I see, thank you for the reviews! One problem that I'm running into testing this change locally is that when the constructor is passed an `Mobject` instance, it does not throw a `TypeError`. 
The test is located in `(tests/ ... /test_vectorized_mobject.py:77)`. 
These are the changes I'm testing:
```python
if isinstance(mob, VMobject):
    flattened_args.append(mob)
elif isinstance(mob, Iterable):
    for submob in mob:
        if not isinstance(submob, VMobject):
            raise TypeError(...)
        flattened_args.extend(mob)
else:
    raise TypeError(...)
```
The problem is that an empty `Mobject` is iterable but, since it is empty, it has no `submob`s and so the constructor never throws a `TypeError`. My first thought was to add another check before iterating through the `submob` as such:
```python
elif isinstance(mob, Iterable):
    if (mob, is instance(Mobject)):
        raise TypeError(...)
    for submob in mob:
        if not isinstance(submob, VMobject):
            raise TypeError(...)
        flattened_args.extend(mob)
```
but, going off what you've already mentioned, I take it that it's not the best solution because it adds yet another copy of the TypeError and it is less readable. How do you think we could solve this without degrading the code's readability?

Reply:
Nice catch, didn't think about that. It could be circumvented with something like this
```py
if isinstance(mob, VMobject):
    ...
elif isinstance(mob, Iterable) and type(mob) is not Mobject:
    for submob in mob:
        if not isinstance(submob, VMobject):
            raise
    flattened.extend(mob)
else:
    raise
```
Not as clean as I like, but it should keep only two copies of the error message.

I would also suggest putting the error message into a function and simply call that function both times to reduce the repetition of the error.

Thanks for your patience :)